advanced/wallets/react-wallet-v2/src/lib/smart-accounts/KernelSmartAccountLib.ts

```diff
@@ -94,27 +122,89 @@ export class KernelSmartAccountLib implements EIP155Wallet {
       type: this.type
     })
     this.initialized = true
+
+    // just testing
+    console.log(
+      'test permissionsContext:',
+      await this.getPermissionsContext('0x957c92075bB364B66560A89e141637BFBda76d5a', BigInt(100))
+    )
+  }
+
+  async getPermissionsContext(sessionPublicKey: Address, limit: bigint): Promise<Hex> {
+    const donutValidator = await signerToDonutValidator(this.publicClient!, {
+      signer: addressToEmptyAccount(sessionPublicKey),
+      donutLimit: limit,
+      validatorAddress: DONUT_VALIDATOR_ADDRESS
+    })
+
+    const executeSelector = toFunctionSelector(
+      getAbiItem({ abi: KernelV3ExecuteAbi, name: "execute" })
+    )
+
+    const account = await createKernelAccount(this.publicClient!, {
+      entryPoint: ENTRYPOINT_ADDRESS_V07,
+      plugins: {
+        sudo: this.validator,
+        regular: donutValidator,
+        entryPoint: ENTRYPOINT_ADDRESS_V07,
+        executorData: {
+          selector: executeSelector,
+          executor: zeroAddress
+        }
+      }
+    })
+
+    console.log('factory:', await account.getFactory())
+    console.log('factory data:', await account.getFactoryData())
+
+    const enableSig = await account.kernelPluginManager.getPluginEnableSignature(account.address)
+
+    return encodePacked(
+      ['bytes1', 'bytes1', 'address', 'address', 'bytes'],
+      [
+        '0x01',
+        '0x01',
+        DONUT_VALIDATOR_ADDRESS,
+        zeroAddress,
+        encodeAbiParameters(
+          parseAbiParameters('bytes, bytes, bytes, bytes'),
+          [
+            encodePacked(
+              ['address', 'uint256'],
+              [sessionPublicKey, limit],
+            ),
+            '0x',
+            executeSelector,
```

Comment:
this has to be
encodePacked(['bytes4', 'address', 'address'], [executeSelector, address(0), address(0)]
)